src/@ionic-native/plugins/intel-security/index.ts

```diff
@@ -0,0 +1,248 @@
+ * - Establish a connection with remote server (E.g. XHR) using a protected channel. SSL/TLS establishment and usage will be done for you by the API implementation
+ *
+ * For more information please visit the [API documentation](https://software.intel.com/en-us/app-security-api/api).
+ *
+ * @usage
+ * ```
+ * import { IntelSecurity } from '@ionic-native/intel-security';
+ * ...
+ * constructor(private intelSecurity: IntelSecurity) { }
+ * ...
+ *
+ * let storageID = 'id';
+ *
+ * this.intelSecurity.data.createFromData({data: 'Sample Data'})
+ *   .then((instanceID: any) => this.IntelSecurity.storage.write(id, instanceID))
+ *   .catch((error: any) => console.log(error));
+ *
+ * this.intelSecurity.storage.read(storageID)
+ *   .then(this.intelSecurity.data.getData)
+ *   .then((data: string) => console.log(data)) // Resolves to 'Sample Data'
+ *   .catch((error: any) => console.log(error));
+ *
+ * this.intelSecurity.storage.delete(storageID)
+ *   .then(() => console.log('Deleted Successfully'))
+ *   .catch((error: any) => console.log(error));
+ *
+ * ```
+ */
+@Plugin({
+  pluginName: 'IntelSecurity',
+  plugin: 'com-intel-security-cordova-plugin',
+  pluginRef: 'intel.security',
+  repo: 'https://github.com/AppSecurityApi/com-intel-security-cordova-plugin'
+})
+@Injectable()
+export class IntelSecurity {
+
+  public storage: IntelSecurityStorage;
+  public data: IntelSecurityData;
+
+  constructor() {
+    this.storage = new IntelSecurityStorage();
+    this.data = new IntelSecurityData();
+
+    try {
+      if (!window.intel)
+        throw true;
+    } catch (e) {
+      console.warn('Native: Intel App Security API is not installed or you are running on a browser.');
+    }
+  }
+}
+
+/**
+ * @hidden
+ */
+@Plugin({
+  pluginName: 'IntelSecurity',
+  plugin: 'com-intel-security-cordova-plugin',
+  pluginRef: 'intel.security.secureData',
+  repo: ''
+})
+export class IntelSecurityData {
+
+  /**
+  * This creates a new instance of secure data using plain-text data.
+  * @param options {Object}
+  * @param options.data {String} Non-empty string.
+  * @param [options.tag] {String} Tag text.
+  * @param [options.extraKey] {Number} Valid secure data instance ID.
+  * @param [options.appAccessControl] {Number} Application access control policy.
+  * @param [options.deviceLocality] {Number} Device locality policy.
+  * @param [options.sensitivityLevel] {Number} Sensitivity level policy.
+  * @param [options.noStore] {Boolean} Disallow sealed blob access.
+  * @param [options.noRead] {Boolean} Disallow plaintext data access.
+  * @param [options.creator] {Number} Creator unique ID.
+  * @param [options.owners] {Number[]} Array of owners unique ID.
+  * @param [options.webOwners] {String[]} List of trusted web domains.
+  * @returns {Promise<any>} Returns a Promise that resolves with the instanceID of the created data instance, or rejects with an error.
+  */
+  @Cordova()
```

Comment:
Replace all `@Cordova()` decorators with `@Cordova({ otherPromise: true })`.

I just realized their plugin returns a promise instead of using callbacks. This decorator config will convert their `Q` promise to a regular ES6 Promise.